### PR TITLE
Chat width via props

### PR DESCRIPTION
**Pull request by danmarshall**
Max width of carousel is determined by the History and then passed down as props.
Resize behavior is now an option which may be either 'none', 'window' or 'detect'.

### CODE REVIEW

**Comment by billba** [comment]
Phew!

File: src/Carousel.tsx [hunk 1]
```diff
@@ -23,14 +24,14 @@ export class Carousel extends React.Component<CarouselProps, CarouselState> {
...
-    private resizeListener = () => this.resize();
-    private scrollEventListener =() => this.onScroll();
+    private scrollEventListener = () => this.onScroll();
     private scrollAllowInterrupt = true;
 
     constructor(props: CarouselProps) {
         super(props);
 
         this.state = {
+            contentWidth: undefined,
```
Comment (MoisesFlores22) [inline src/Carousel.tsx]: I merge this pr to my master branch and is broken @billba

Comment (danmarshall) [inline src/Carousel.tsx]: Can you provide the build error?

Comment (MoisesFlores22) [inline src/Carousel.tsx]: screen shot 2017-03-13 at 12 20 18 pm

Comment (billba) [inline src/Carousel.tsx]: You need to `npm install` to get the latest version of TypeScript and the latest React type declarations.

Comment (MoisesFlores22) [inline src/Carousel.tsx]: thanks

**Pull request merged**